packages/core/snapshot/read.go

```diff
@@ -85,68 +78,129 @@ func readSnapshotHeader(reader io.ReadSeeker) (*ledger.SnapshotHeader, error) {
 	}
 	header.DiffEpochIndex = epoch.Index(index)
 
+	var latestECRecordLen int64
+	if err := binary.Read(reader, binary.LittleEndian, &latestECRecordLen); err != nil {
+		return nil, fmt.Errorf("unable to read latest ECRecord bytes len: %w", err)
+	}
+
+	ecRecordBytes := make([]byte, latestECRecordLen)
+	if err := binary.Read(reader, binary.LittleEndian, ecRecordBytes); err != nil {
+		return nil, fmt.Errorf("unable to read latest ECRecord: %w", err)
+	}
+	header.LatestECRecord = &epoch.ECRecord{}
+	if err := header.LatestECRecord.FromBytes(ecRecordBytes); err != nil {
+		return nil, err
+	}
+
 	return header, nil
 }
 
-func readSolidEntryPoints(scanner *bufio.Scanner) (seps *SolidEntryPoints, err error) {
-	scanner.Scan()
-	data := scanner.Bytes()
+func readSolidEntryPoints(reader io.ReadSeeker) (seps *SolidEntryPoints, err error) {
+	seps = &SolidEntryPoints{}
+	blkIDs := make([]tangleold.BlockID, 0)
+
+	// read seps EI
+	var index int64
+	if err := binary.Read(reader, binary.LittleEndian, &index); err != nil {
+		return nil, fmt.Errorf("unable to read fullEpochIndex: %w", err)
+	}
+	seps.EI = epoch.Index(index)
+
+	// read numbers of solid entry point
+	var sepsLen int64
+	if err := binary.Read(reader, binary.LittleEndian, &sepsLen); err != nil {
+		return nil, fmt.Errorf("unable to read seps len: %w", err)
+	}
+
+	spentLenInt := int(sepsLen)
+	for i := 0; i < spentLenInt; {
+		var sepsBytesLen int64
+		if err := binary.Read(reader, binary.LittleEndian, &sepsBytesLen); err != nil {
+			return nil, fmt.Errorf("unable to read seps bytes len: %w", err)
+		}
+
+		sepsBytes := make([]byte, sepsBytesLen)
+		if err := binary.Read(reader, binary.LittleEndian, sepsBytes); err != nil {
+			return nil, fmt.Errorf("unable to read solid entry points: %w", err)
+		}
 
-	if len(data) > 0 {
-		seps = &SolidEntryPoints{}
-		if _, err = serix.DefaultAPI.Decode(context.Background(), data, &seps, serix.WithValidation()); err != nil {
+		ids := make([]tangleold.BlockID, 0)
+		_, err = serix.DefaultAPI.Decode(context.Background(), sepsBytes, &ids, serix.WithValidation())
+		if err != nil {
 			return nil, err
 		}
+		blkIDs = append(blkIDs, ids...)
+		i += len(ids)
 	}
 
+	seps.Seps = blkIDs
+
 	return seps, nil
 }
 
 // readOutputWithMetadata consumes a slice of OutputWithMetadata from the given reader.
-func readOutputWithMetadata(scanner *bufio.Scanner) (outputMetadatas []*ledger.OutputWithMetadata, err error) {
-	scanner.Scan()
-	data := scanner.Bytes()
+func readOutputWithMetadata(reader io.ReadSeeker) (outputMetadatas []*ledger.OutputWithMetadata, err error) {
```

Comment:
```suggestion
func readOutputsWithMetadata(reader io.ReadSeeker) (outputMetadatas []*ledger.OutputWithMetadata, err error) {
```

Reply:
Is this method actually to consume multiple outputs? You seem to use it to consume single outputs.

Reply:
No, it reads a slice of OutputsWithMetadata, I renamed it to readOutputsWithMetadata`s`